packages/utils/tool-utils/src/odspTokenManager.ts

```diff
@@ -128,19 +128,28 @@ export class OdspTokenManager {
         cacheKey: IOdspTokenManagerCacheKey,
     ) {
         const memoryCache = cacheKey.isPush ? this.pushCache : this.storageCache;
-        const memoryToken = memoryCache.get(cacheKey.server);
+        const memoryToken = memoryCache.get(cacheKey.userOrServer);
         if (memoryToken) {
             debug(`${cacheKeyToString(cacheKey)}: Token found in memory `);
             return memoryToken;
         }
         const fileToken = await this.tokenCache?.get(cacheKey);
         if (fileToken) {
             debug(`${cacheKeyToString(cacheKey)}: Token found in file`);
-            memoryCache.set(cacheKey.server, fileToken);
+            memoryCache.set(cacheKey.userOrServer, fileToken);
             return fileToken;
         }
     }
 
+    private static getCacheKey(
+        isPush: boolean,
+        tokenConfig: OdspTokenConfig,
+        server: string,
+    ): IOdspTokenManagerCacheKey {
+        // If we are using password, we should cache the token per user instead of per server
+        return { isPush, userOrServer: tokenConfig.type === "password" ? tokenConfig.username : server };
```

Comment:
I do not have better suggesting, but having "userOrServer" property seems a bit weird.

Reply:
yep. I couldn't think of something better, other than a generic name like "key", which I think is worse.